File: PDF2_Ex2.py
#NÃO TESTADO, FUNCIONA NOS PRIMEIROS NÚMEROS (TESTES RETORNAM RAPIDO PELO MENOS ATÉ <=112) MAS NAO FOI TESTADO COM <=500
soma = 0
divisores = 0
cont = 1
while(divisores<=500):
	divisores = 1	#Todo numero é divisivel por 1
	soma = soma + cont
	x=soma
	b=1
	while(int(x)>1):
		if(x == int(x)):
			divisores+=1
		b+=1
		x = soma/b
	cont+=1
print(soma)
print(cont)

File: PDF2_Ex2.c
#include<stdio.h>

int main(){
	int soma = 0, qtddivisores = 0, cont = 0, divisor = 1;
	while(qtddivisores<=500){
		qtddivisores = 1; //Todo número é divisivel por si próprio.
		divisor = 1;
		cont++;
		soma = soma + cont;           //Soma dos números naturais.
		while(divisor<=(soma/2)+1){   //Vai até a metade da soma pois o maior numero que divide um
			if(soma%divisor==0){        //número qualquer sem ser ele mesmo é a sua metade
				qtddivisores++;
			}
			divisor++;
		}
	}
	printf("%d\n", soma);
	return 0;
}
